src/modules/twap/state/twapOrdersListAtom.ts

```diff
@@ -0,0 +1,33 @@
+import { atom } from 'jotai'
+import { atomWithStorage } from 'jotai/utils'
+
+import { tokensByAddressAtom } from 'modules/tokensList/state/tokensListAtom'
+import { walletInfoAtom } from 'modules/wallet/api/state'
+
+import { TwapOrderItem } from '../types'
+import { emulateTwapAsOrder } from '../utils/emulateTwapAsOrder'
+
+export type TwapOrdersList = { [key: string]: TwapOrderItem }
+
+export const twapOrdersListAtom = atomWithStorage<TwapOrdersList>('twap-orders-list:v2', {})
+
+export const updateTwapOrdersListAtom = atom(null, (get, set, nextState: TwapOrdersList) => {
+  const currentState = get(twapOrdersListAtom)
+
+  if (JSON.stringify(nextState) !== JSON.stringify(currentState)) {
```

Comment:
Is there any chance the state sorting could be different?
In other words, JS doesn't guarantee the obj sort order, does it?

Reply:
In my head, I had thought "JS doesn't guarantee the order".
But, there are some rules: https://blog.devgenius.io/are-javascript-object-properties-ordered-c30597754e5c
I understand, this is not the best solution to avoid re-renders, I have one more idea:
https://dmitripavlutin.com/how-to-compare-objects-in-javascript/
So, I will implement custom functions to compare these collections, it will be at least better from the performance perspective.
Thank you!

Reply:
p.s. The solution with `JSON.stringify` works properly because we build states always in the same order, because we get data from Safe API and from the tokens list in the same order always.

Reply:
Wow, nodejs has native implementation for `deepEqual`:
https://www.w3schools.com/nodejs/met_assert_deepequal.asp

Reply:
The fix: https://github.com/cowprotocol/cowswap/pull/2597/commits/5e51586e6dcdea36089dde5f82d08ae5294afd07

Reply:
Small comment https://github.com/cowprotocol/cowswap/commit/5e51586e6dcdea36089dde5f82d08ae5294afd07#r117051119

In summary, we might already have deep equal indirect dependencies, might as well use them instead 🤷

Reply:
Awesome! Thanks! Used it